packages/api-codec/src/codec/EnumType.ts

```diff
@@ -52,7 +52,14 @@ export default class EnumType <T> extends Base<Base<T>> {
     return this;
   }
 
-  setValue (index?: number, value?: any): void {
+  setValue (index?: | EnumType<T> | number, value?: any): void {
+    if (index instanceof EnumType) {
```

Comment:
Not a fan of this... it would have been fine if it was value, however value/index are most probably used wrongly, index is the actual value, value is well, the constructor value.